app/src/main/java/com/cmput301f20t21/bookfriends/ui/profile/ProfileEditDialog.java

```diff
@@ -0,0 +1,105 @@
+import androidx.fragment.app.DialogFragment;
+
+import com.cmput301f20t21.bookfriends.R;
+
+import org.jetbrains.annotations.NotNull;
+
+public class ProfileEditDialog extends DialogFragment {
+    private EditText editEmail;
+    private EditText editPhone;
+    private TextView cancel;
+    private TextView confirm;
+    private TextView existEmail;
+    private TextView existPhone;
+    public ProfileEditListener listener;
+
+    @NotNull
+    @Override
+    public View onCreateView(final LayoutInflater inflater, @Nullable ViewGroup container,Bundle savedInstanceState) {
+        View view = inflater.inflate(R.layout.dialog_editprofile,container,false);
+        editEmail = view.findViewById(R.id.edit_email);
+        editPhone = view.findViewById(R.id.edit_phone);
+        cancel = view.findViewById(R.id.text_cancel);
+        confirm = view.findViewById(R.id.text_confirm);
+        existEmail = view.findViewById(R.id.email);
+        existPhone = view.findViewById(R.id.phone);
+
+        getDialog().getWindow().setBackgroundDrawableResource(R.drawable.button_round_corner);
+        cancel.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                getDialog().dismiss();
+            }
+        });
+        confirm.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                String inputEmail = editEmail.getText().toString();
+                String inputPhone = editPhone.getText().toString();
+                boolean v1 = isEmailValid(inputEmail);
+                boolean v2 = isPhoneValid(inputPhone);
+                if (v1 && v2){
+                    listener.editing(inputEmail, inputPhone);
+                    getDialog().dismiss();
+                }
+            }
+        });
+        return view;
+
+    }
+
+   @Override
+    public void onAttach(@NonNull Context context) {
+        super.onAttach(context);
+        try{
+            listener = (ProfileEditListener)getTargetFragment();
+        }catch (ClassCastException e){
+           throw new ClassCastException("Need Implement ProfileEditListener");
+        }
+    }
+
+    public interface ProfileEditListener{
+        void editing(String email, String phone);
+    }
+
+    // check if entered email address is valid or not
+    // if not, show message
+    boolean isEmailValid(CharSequence email) {
+        boolean valid =  android.util.Patterns.EMAIL_ADDRESS.matcher(email).matches();
+        if (!valid && email.length()!= 0){
+            editEmail.setError("Email address is INVALID.");
+            return false;
+        }
+        else{
+            editEmail.setError(null);
+            return true;
+        }
+    }
+    /// check if entered phone number is valid or not
+    // if not, show message
+    boolean isPhoneValid(String phone) {
+        if (phone.length()!=10 && phone.length()!=0){
```

Comment:
I think android.util.Pattern have [phone matcher](https://developer.android.com/reference/android/util/Patterns#PHONE) as well, we might just use that for now, hardcoding 10 seems weird since in some country (e.g. China) the phone number is 11 digits and we might also want country code included in the phone number like +17801234567